wp-includes/rest-api/endpoints/class-wp-rest-customize-panels-controller.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * REST API: WP_REST_Customize_Panels_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since ?.?.?
+ */
+
+/**
+ * Core class to access posts via the REST API.
+ *
+ * @since ?.?.?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Customize_Panels_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 4.7.0
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'customize/v1';
+		$this->rest_base = 'panels';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<panel>[\w-]+)', array(
+			'args' => array(
+				'panel' => array(
+					'description' => __( 'An alphanumeric identifier for the panel.' ),
+					'type'        => 'string',
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => array(
+					'context' => $this->get_context_param( array(
+						'default' => 'view',
+					) ),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Retrieves the panel's schema, conforming to JSON Schema.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/schema#',
+			'title'      => 'panel',
+			'type'       => 'object',
+			'properties' => array(
+				'description'     => array(
+					'description' => __( 'Panel description.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'name'            => array(
+					'description' => __( 'The title for the panel.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'priority'        => array(
+					'description' => __( 'The priority of the panel.' ),
+					'type'        => 'integer',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'sections'        => array(
+					'description' => __( 'The sections of the panel.' ),
+					'type'        => 'object',
```

Comment:
Currently implemented that version ^, let me know if that's not optimal, happy to change. Here is a sample output as of current implementation:
```
{
    "type": "nav_menus",
    "id": "nav_menus",
    "priority": 700,
    "theme_supports": "",
    "title": "Menus",
    "description": "<p>This panel is used for managing navigation menus for content you have already published on your site. You can create menus and add items for existing content such as pages, posts, categories, tags, formats, or custom links.</p><p>Menus can be displayed in locations defined by your theme or in <a href=\"javascript:wp.customize.panel( 'widgets' ).focus();\">widget areas</a> by adding a &#8220;Custom Menu&#8221; widget.</p>",
    "auto_expand_sole_section": false,
    "sections": [
      {
        "section_id": "menu_locations",
        "_link": "customize/v1/sections/menu_locations"
      },
      {
        "section_id": "add_menu",
        "_link": "customize/v1/sections/add_menu"
      }
    ]
  }
```